cdap-app-fabric/src/main/java/io/cdap/cdap/internal/provision/ProvisioningService.java

```diff
@@ -154,10 +158,19 @@
   @Override
   protected void startUp() throws Exception {
     LOG.info("Starting {}", getClass().getSimpleName());
+
+    this.taskExecutor = new KeyedExecutor<>(
+      Executors.newScheduledThreadPool(cConf.getInt(Constants.Provisioner.EXECUTOR_THREADS),
+                                       Threads.createDaemonThreadFactory("provisioning-task-%d")));
+
+    int maxPoolSize = cConf.getInt(Constants.Provisioner.CONTEXT_EXECUTOR_THREADS);
+    ThreadPoolExecutor contextExecutor = new ThreadPoolExecutor(
+      maxPoolSize, maxPoolSize, 60, TimeUnit.SECONDS, new LinkedBlockingQueue<>(),
```

Comment:
Is it safe to have an unbounded queue? Any chance we can run into OOM with it?

Reply:
We typically use unbounded queue for executor with a thread pool (same as the default in Java Executors). If this queue even gets OOM, we are in bigger problem than the OOM itself.